backend/wallet/services/offchain/payment_command.py

```diff
@@ -1,159 +1,84 @@
-# Copyright (c) The Diem Core Contributors
-# SPDX-License-Identifier: Apache-2.0
 import logging
-import typing
+import uuid
 from datetime import datetime
-from typing import Optional, Tuple, Callable, List
+from typing import Optional, Callable, List
 
 import context
-from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
-from diem import offchain, identifier
-from diem.offchain import CommandType
+from diem import offchain, identifier, txnmetadata
+from diem.offchain import Status
 from diem_utils.types.currencies import DiemCurrency
 from wallet import storage
-from wallet.services import account, kyc
-from wallet.storage import models
-from wallet.services.fund_pull_pre_approval import (
-    process_funds_pull_pre_approvals_requests,
-    handle_fund_pull_pre_approval_command,
+from wallet.services import account
+from wallet.services.offchain.utils import (
+    hrp,
+    account_address_and_subaddress,
+    user_kyc_data,
+    compliance_private_key,
+    generate_my_address,
 )
-
-# noinspection PyUnresolvedReferences
-from wallet.storage.funds_pull_pre_approval_command import (
-    get_account_commands,
-    FundsPullPreApprovalCommandNotFound,
-    commit_command,
-    get_commands_by_sent_status,
-    get_command_by_id,
-    update_command,
-    get_account_command_by_id,
-)
-
-from ..storage import (
+from wallet.storage import (
+    save_payment_command,
+    models,
     lock_for_update,
     get_account_id_from_subaddr,
     Transaction,
-    save_payment_command,
-)
-from ..types import (
-    TransactionType,
-    TransactionStatus,
 )
+from wallet.types import TransactionStatus, TransactionType
 
 logger = logging.getLogger(__name__)
-
 PaymentCommandModel = models.PaymentCommand
 
 
-def save_outbound_payment_command(
-    sender_id: int,
-    destination_address: str,
-    destination_subaddress: str,
-    amount: int,
-    currency: DiemCurrency,
-) -> offchain.PaymentCommand:
-    sender_onchain_address = context.get().config.vasp_address
-    sender_sub_address = account.generate_new_subaddress(account_id=sender_id)
+def add_payment_command(
+    account_id,
+    reference_id,
+    vasp_address,
+    merchant_name,
+    action,
+    currency,
+    amount,
+    expiration: int,
+) -> None:
```

Comment:
This method is not adding just any payment command, but only the commands where I am a sender. It should be reflected somehow in the name.